### PR TITLE
Adds a species indicator for med/sec huds (and indirectly ai)

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Adds a species hud icon at the top right of your characters sprite.
The hud is NOT perfect at getting your species, having a mask on will make it check your id's name against the medrecords.
And, setting someone to human in medrecords always makes it show human, regardless of your actual species (for ai)

### Medical
Quickly noting what species a player is helps medical to do their job more efficiently, accounting for species quirks so you dont accidentily poison an oozeling for example.

### Security
Quickly getting more information on a suspect (roboticist lizard in bridge)

### AI/Borg
Helps silicons follow asimov easier, at a glance figuring out who to consider human and helping identify them even under a full spacesuit. This also lets players more easily influence their human-ness to the ai, since non humans can get a human's id or ask med to set them to human. And, humans are under slightly more risk to the ai if they are unrecognizable (mask and no id/forged medrecord)

Screenshots
image

species_use_records

## Changelog

:cl:
add: species hud icons for med/secHUD
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
what will it show if the ID doesn't link up to med records?

**Comment by ShizCalev** [comment]
does this respect changeling memes?

**Comment by etherware-novice** [comment]
> what will it show if the ID doesn't link up to med records? 

hudunknown1, a coincidentilly unused sprite thats in the perfect position for this

**Comment by etherware-novice** [comment]
> does this respect changeling memes? 

if changelings morphing replaces dna then yea

**Comment by EuclidianGeometer** [comment]
isn't this a nerf to the agent ID and the like?

**Comment by etherware-novice** [comment]
good point! do you think making agent huds display human would be a good compromise?

**Comment by msgerbs** [comment]
This somewhat contradicts existing AI policy which is that the AI should treat entirely unidentifiable beings as human until proven otherwise. That's probably not a dealbreaker, but is something to consider.

**Comment by etherware-novice** [comment]
it doesnt say they're *not* human it just cant confirm it is, the ai can still like treat them as human ig

**Comment by msgerbs** [comment]
Definitely. Like I said, probably not a deal breaker. I can see it tripping up a new AI player, but add it to the list I guess. AI is a landmine of bwoinks.

**Comment by etherware-novice** [comment]
hey thats why you need to have..i think 10? hours in ai to play it in the first place :>

**Comment by msgerbs** [comment]
That is fair. In any case I love you for this, I hate when I think somebody's human and they're not and now the crew's mad at me for protecting a cat.

**Comment by etherware-novice** [comment]
youre welcome!

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -943,6 +943,77 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
...
 	usr = temp
 
+/proc/species_name_to_hud(species)
+	species = lowertext(species) // sanitize your text
+	if(species == "human")  // gotta *love* species code!!
+		return "species_human"
+
+	// theres so many repeats here to try and catch every variant for species players would enter
+	var/static/list/lookup = list(
+		"cat" 			= /datum/species/human/felinid,
```
Comment (Mothblocks) [inline code/__HELPERS/mobs.dm:953]: This entire proc seems like garbage to me. Only use the names the code ever actually uses in game.

File: code/game/data_huds.dm [hunk 2]
```diff
@@ -509,3 +509,40 @@ Diagnostic HUDs!
...
+	var/image/holder = hud_list[SPECIES_HUD]
+
+	// to not inder nerf the agent card, make it always show human
+	if(istype(get_idcard(), /obj/item/card/id/advanced/chameleon))
+		holder.icon_state = "species_human"
+		return
+
+	var/perpname = get_face_name(get_id_name(""))
+	if(perpname && GLOB.data_core)
+		var/datum/data/record/R = find_record("name", perpname, GLOB.data_core.medical)
```
Comment (Mothblocks) [inline code/game/data_huds.dm:524]: Full name--`record`.

File: code/game/data_huds.dm [hunk 3]
```diff
@@ -509,3 +509,40 @@ Diagnostic HUDs!
 	var/image/holder = hud_list[DIAG_AIRLOCK_HUD]
 	holder.icon_state = "electrified"
 	set_hud_image_active(DIAG_AIRLOCK_HUD)
+
+// Species info!
+/mob/living/carbon/human/proc/update_species_hud(species_overide)
+	var/image/holder = hud_list[SPECIES_HUD]
+
+	// to not inder nerf the agent card, make it always show human
```
Comment (Mothblocks) [inline code/game/data_huds.dm:517]: "inder"?

File: code/game/data_huds.dm [hunk 4]
```diff
@@ -509,3 +509,40 @@ Diagnostic HUDs!
...
+	if(istype(get_idcard(), /obj/item/card/id/advanced/chameleon))
+		holder.icon_state = "species_human"
```
Comment (Mothblocks) [inline code/game/data_huds.dm:519]: I don't understand how this is an ideal path considering not all traitors are human?

Comment (Mothblocks) [reply code/game/data_huds.dm:519]: Shouldn't you be able to customize it or something?

File: code/game/data_huds.dm [hunk 5]
```diff
@@ -509,3 +509,40 @@ Diagnostic HUDs!
 	var/image/holder = hud_list[DIAG_AIRLOCK_HUD]
 	holder.icon_state = "electrified"
 	set_hud_image_active(DIAG_AIRLOCK_HUD)
+
+// Species info!
+/mob/living/carbon/human/proc/update_species_hud(species_overide)
```
Comment (Mothblocks) [inline code/game/data_huds.dm:514]: override, not overide

File: code/game/data_huds.dm [hunk 6]
```diff
@@ -509,3 +509,40 @@ Diagnostic HUDs!
 	var/image/holder = hud_list[DIAG_AIRLOCK_HUD]
 	holder.icon_state = "electrified"
 	set_hud_image_active(DIAG_AIRLOCK_HUD)
+
+// Species info!
```
Comment (Mothblocks) [inline code/game/data_huds.dm:513]: This is not good documentation

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I am not really convinced this is a useful feature at all and it is going to conflict with the antag HUD, which is shown in game to cultists, revs, etc, which is going to look really bad.

**Comment by Mothblocks** [comment]
Also wait I love AIs not realizing a human is a cat and vice versa

**Comment by Mothblocks** [comment]
I'm going to give this time to simmer in my head but I don't think I like this at all, giving this some time for maintainers to say if this is something they're interested in

**Comment by optimumtact** [comment]
mothblocks just back yourself

**Comment by msgerbs** [comment]
Sad

**Comment by etherware-novice** [comment]
i have the patch file saved if i want to reopen at some pt

**Pull request closed without merging**